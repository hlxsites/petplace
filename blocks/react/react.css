:root{--highlight-color: #ff7d5a;--cta-button-color: #c74d2f;--compare-button-bg-color: var(--cta-button-color);--number-background-color: #7369ab;--faq-background-color: #f1f0f7;--faq-accordion-txt-color: #2f2f33;--faq-accordion-arrow-bg-color: #6e6d73;--figo-powered-txt-color: #585973;--pet-insurance-txt-color: #09090d;--error-message-bg-color: #fcefed;--error-message-color: #ba271a;--link-color: var(--cta-button-color);--link-hover-color: var(--highlight-color);--link-color-alt: var(--color-blue);--link-hover-color-alt: var(--text-color-alt);--background-color: #fff;--background-color-light: #f2f6fc;--background-color-dark: #3b092c;--background-inactive: #F4F4F5;--border-base-color: #E4E4EB;--border-color: #D9D9D9;--border-color-strong: #d0d0d6;--text-color: #28061e;--text-color-inverted: #fff;--text-color-alt: #363636;--text-color-disabled:#464646;--text-color-subtle: #696770;--text-color-accent: #3c0040;--text-color-supporting: #2f2f33;--color-gray: #8f8f8f;--color-gray-light: #b9c2cc;--color-gray-light-alt: #e4e4e4;--color-gray-lighter: #f7f7f7;--color-gray-lighter-alt: #BDBDBD;--color-gray-lightest: #f9f9f9;--color-gray-lighest-alt: #E7E7E7;--color-gray-dark: #4a4a4a;--color-black: #000;--color-blue: #678fda;--color-blue-light: #c4d1e9;--color-blue-dark: #5773a8;--color-blue-dark-alt: #5D73A4;--color-purple: #6f69a8;--color-purple-alt: #6E6AA4;--color-purple-light: #978ce0;--color-purple-gray: #736aa4;--color-purple-light-alt: #8e8199;--color-purple-dark: #302280;--color-purple-dark-alt: #440042;--color-purple-300: #7369ab;--color-purple-500: #3c3759;--color-dark-maroon: #5d3450;--color-maroon-300: #3C0040;--color-red: #ff6662;--color-gold: #d8c46c;--color-green: #97baa7;--color-magenta: #bc6da6;--color-mauve: #8c4877;--color-orange: var(--highlight-color);--color-orange-transparent: rgb(255 85 39 / 60%);--color-blue-transparent: rgb(62 113 208 / 60%);--color-blue-dark-transparent: rgb(70 92 134 / 60%);--color-purple-transparent: rgb(87 81 141 / 60%);--color-purple-light-transparent: rgb(115 100 213 / 60%);--color-red-transparent: rgb(255 52 47 / 60%);--color-gold-transparent: rgb(205 180 68 / 60%);--color-green-transparent: rgb(120 166 141 / 60%);--color-magenta-transparent: rgb(169 77 143 / 60%);--color-mauve-transparent: rgb(106 55 90 / 60%);--color-neutral-300: #e4e4e7;--color-neutral-600: #6e6d73;--body-font-family: "Libre Franklin", "Libre Franklin Fallback", sans-serif;--heading-font-family: "Raleway", "Raleway Fallback", sans-serif;--fixed-font-family: "Roboto Mono", menlo, consolas, "Liberation Mono", monospace;--body-font-size-default: 16px;--body-font-size-xl: 22px;--body-font-size-l: 20px;--body-font-size-m: var(--body-font-size-default);--body-font-size-s: 14px;--body-font-size-xs: 13px;--body-font-size-xxs: 10px;--heading-font-size-xxxl: 3.222em;--heading-font-size-xxl: 2.1em;--heading-font-size-xl: 1.75em;--heading-font-size-l: 1.222em;--heading-font-size-m: 1.125em;--heading-font-size-s: 1em;--heading-font-size-xs: .875em;--heading-font-size-xxs: .75em;--elevation-height-xs: 0 1px 3px 1px rgba(0 0 0 / .15), 0 1px 2px 0 rgba(0 0 0 / .3);--nav-height: 64px;--card-details-mask: linear-gradient(to bottom, #fff 44.33%, rgba(255, 255, 255, 88%) 67.03%, rgba(255, 255, 255, 0%) 100.9%)}@media (min-width: 600px){:root{--body-font-size-default: 18px;--body-font-size-s: 16px;--body-font-size-xs: 14px;--body-font-size-xxs: 12px;--heading-font-size-xxl: 2.75rem}}@media (min-width: 900px){:root{--heading-font-size-xxl: 3.125rem;--heading-font-size-xl: 2em}}@media (min-width: 1024px){:root{--nav-height: 147px}}@media (min-width: 1216px){:root{--heading-font-size-xxl: 4.125rem;--heading-font-size-xl: 2em}}@font-face{font-family:Libre Franklin Fallback;size-adjust:103.331%;src:local("Arial")}@font-face{font-family:Raleway Fallback;size-adjust:107.941%;src:local("Arial")}:focus-visible{outline:solid 3px #639af9;outline-offset:5px;visibility:visible;opacity:1;outline-style:inset}html{scrollbar-gutter:stable}body{font-size:var(--body-font-size-default);margin:0;min-width:300px;font-family:var(--body-font-family);line-height:1.5;color:var(--text-color);background-color:var(--background-color);display:none;-moz-osx-font-smoothing:gray;-webkit-font-smoothing:antialiased;text-rendering:optimizelegibility}body.appear{display:block}header{height:var(--nav-height)}h1,h2,h3,h4,h5,h6{font-family:var(--heading-font-family);font-weight:600;line-height:1.25;margin-top:1.5rem;margin-bottom:1rem;scroll-margin:calc(var(--nav-height) + 1em);text-wrap:balance}h3,h4,h5,h6{margin-top:0}h1{font-size:var(--heading-font-size-xxl);color:var(--text-color-inverted);line-height:1.1}h2{font-size:var(--heading-font-size-xl)}h3{font-size:var(--heading-font-size-l);color:var(--text-color-alt)}h4{font-size:var(--heading-font-size-m)}h5{font-size:var(--heading-font-size-s)}h6{font-size:var(--heading-font-size-xs)}:is(h1,h2,h3,h4,h5,h6) em{color:var(--highlight-color);font-style:normal}@media (min-width: 600px){h1,h2,h3,h4,h5,h6{text-wrap:unset}}p,dl,ol,ul,pre,blockquote{margin:0}strong{color:var(--text-color-alt)}hr{margin-top:1.5em;margin-bottom:1.5em;border:0;border-bottom:1px solid var(--background-color-light)}code,pre,samp{font-family:var(--fixed-font-family);font-size:var(--body-font-size-s)}pre{overflow:scroll}ol{display:block;padding-left:1.5rem;padding-right:1.5rem;font-family:var(--body-font-family)}li{list-style:inherit;list-style-position:outside;padding-left:.5rem;font-size:var(--body-font-size-default);line-height:1.5;font-weight:400}main pre{background-color:var(--background-color-light);padding:1em;border-radius:.25em;overflow:auto;white-space:pre}main a{color:var(--link-color);text-decoration:none}main a:hover{text-decoration:none}main a strong{color:inherit}.button-container{display:flex;gap:0 1rem;flex-wrap:wrap}a.button:-moz-any-link{font-family:var(--heading-font-family);display:inline-block;box-sizing:border-box;text-decoration:none;border:2px solid transparent;padding:5px 30px;text-align:center;font-style:normal;font-weight:800;cursor:pointer;color:var(--background-color);background-color:var(--link-color);margin:16px 0;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;border-radius:30px}a.button:any-link,button{font-family:var(--heading-font-family);display:inline-block;box-sizing:border-box;text-decoration:none;border:2px solid transparent;padding:5px 30px;text-align:center;font-style:normal;font-weight:800;cursor:pointer;color:var(--background-color);background-color:var(--link-color);margin:16px 0;white-space:nowrap;overflow:hidden;text-overflow:ellipsis;border-radius:30px}a.button:hover,a.button:focus,button:hover,button:focus{background-color:var(--link-hover-color);cursor:pointer}a.button.silent,button.button.silent{background:inherit;border:none;color:inherit;font:inherit;margin:0;padding:0}button:disabled,button:disabled:hover{background-color:var(--background-color-light);cursor:unset}.dark button:disabled,.dark button:disabled:hover{background-color:var(--color-gray)}a.button.cta,button.cta{padding:.75rem 3rem;border:5px solid var(--background-color);box-shadow:0 4px 30px #0000001a;font-size:var(--heading-font-size-m)}a.button.secondary,button.secondary{background-color:unset;border:2px solid currentcolor;color:var(--text-color)}main ul ul{list-style-type:disc}main input{font-size:1.25rem;width:100%;max-width:50rem;display:block;margin-bottom:1rem;padding:.75rem .6rem;border-radius:.25rem;box-sizing:border-box;border:1px solid var(--text-color);color:var(--text-color);background-color:var(--background-color)}main input:hover{border:1px solid var(--text-color)}main blockquote{font-style:italic;margin:3rem;text-indent:-1rem;hanging-punctuation:first}main blockquote p:before{content:"“";line-height:0}main blockquote p:after{content:"”";line-height:0}main img{display:block;max-width:100%;height:auto}main iframe[src*="youtube.com"]{width:100%;height:auto;aspect-ratio:16 / 9}.icon{display:inline-block;height:24px;width:24px}.icon svg{height:100%;width:100%}main .section{padding:2rem 1.5rem}main .default-content-wrapper~.default-content-wrapper{margin-top:3.75rem}main .section.well{display:flex;flex-direction:column;background:var(--background-color);gap:44px}main .section.well h2{margin:0 0 1.25rem;color:var(--highlight-color)}main .section.well.dark h2{color:inherit}main .section.well li{margin-bottom:1rem;font-family:var(--heading-font-family);font-size:1.25rem;color:#353534;text-align:left;overflow:hidden}main .section.well li::marker{color:var(--highlight-color)}main .section.well .default-content-wrapper{box-sizing:border-box;padding:3.75rem 1.25rem;background:var(--background-color)}main .section.well.dark .default-content-wrapper{padding:0;background-color:inherit}main .section.well .default-content-wrapper>*{margin-left:auto;margin-right:auto;max-width:1000px}main .section.well .default-content-wrapper~.default-content-wrapper{margin-top:0}main .section.well.small{box-sizing:border-box;background-color:var(--background-color-light);max-width:650px;padding:60px;border-radius:.5rem}main .section.well.dark{color:var(--text-color-inverted)}main .section.well.dark.small{background-color:var(--background-color-dark)}main .section.well.dark .icon{margin:0 .5rem;color:var(--text-color-inverted)}main .section[data-background]{position:relative}main .section.highlight[data-background]{mix-blend-mode:darken}main .section.columns-container[data-background] .columns{margin-bottom:0}main .section[data-background] .section-background{position:relative;width:calc(100% + 48px);background-repeat:no-repeat;background-size:cover;aspect-ratio:2 / 1;margin:0 -24px -61px;background-position:center}main .section.well.small[data-background] .section-background{width:calc(100% + 122px);margin:0 -61px -61px}@media (min-width: 900px){main .section[data-background] .section-background{position:absolute;left:50%;bottom:0;width:100vw;height:100%;max-width:none;margin:0;z-index:-1;translate:-50%;background-size:1920px auto;background-position:bottom}main .section.well.small[data-background] .section-background{width:100vw;margin:0}}[data-align=left]{text-align:left}[data-align=center]{text-align:center}[data-align=right]{text-align:right}.icon-user{color:#5773a9;height:30px;width:30px;margin-right:10px}.icon-user circle{color:#5773a9}.icon-user path{color:#c5d2e9}.generic :is(h1,h2,h3,h4,h5,h6){margin-top:1.5rem;font-weight:800;color:inherit}.generic main .section{margin:0 auto;padding:3rem 1.5rem}.generic main .hero-container{padding:0}.generic main .hero{min-height:350px}.generic main p{margin:1rem 0}.generic main a{color:var(--link-color-alt)}.generic main ul{padding-left:rem}.generic main li::marker{color:var(--highlight-color)}.generic a.button{margin-top:0}.generic .columns{margin:60px auto}.generic .columns>div{align-items:center;gap:1.5rem}.generic .blade{gap:3.5rem}.generic .blade>div{gap:3rem}.generic .blade .blade-body :is(h3,h4){font-family:var(--heading-font-family);font-size:2rem;letter-spacing:0;text-transform:none}.generic main a .icon{color:var(--highlight-color);height:2.5rem;width:2.5rem;margin-right:.5rem}.generic main .well{margin:3.75rem auto;font-size:var(--body-font-size-xl)}.generic main .form label{display:none}.generic main .section.well.dark h2:after,.generic main .section.highlight .columns h3:after{content:"";display:block;width:1.5rem;height:2px;background:var(--highlight-color);margin:1rem auto}.generic main .section.highlight .columns h3{font-size:2rem}@media (min-width: 600px){.generic h1{font-size:2.667em}.generic main h2{font-size:1.75em}}@media (min-width: 900px){.generic h1{font-size:3.222em}.generic main .centered h2{font-size:2.5em;margin-bottom:60px}.generic main .columns h2{font-size:2em}}main .centered{text-align:center}main .centered .button-container{justify-content:center}main .section.highlight{margin:0 auto;background-color:var(--background-color-light)}main .section.highlight.strong{background-color:var(--background-color-dark);color:var(--text-color-inverted)}main .section.highlight.purple{background-color:var( --color-purple-dark);color:var(--text-color-inverted)}[aria-hidden=true],[aria-expanded=false]~[role=group]{display:none}.sr-focusable{position:absolute;top:var(--nav-height)}.sr-focusable a,a.sr-focusable{position:absolute;opacity:0;transform:translateY(-100%);white-space:nowrap;z-index:-1}.sr-focusable a:active,.sr-focusable a:focus,a.sr-focusable:active,a.sr-focusable:focus{z-index:9999;padding:.1875rem;background-color:var(--color-delta-red);color:var(--text-color-inverted);font-family:var(--body-medium-font-family);font-size:var(--body-font-size-s);opacity:1;text-transform:uppercase;transform:none}.sr-only:not(:focus,:active){clip:rect(0 0 0 0);clip-path:inset(50%);height:1px;overflow:hidden;position:absolute;white-space:nowrap;width:1px}.a11y-quicknav{position:fixed;margin:0;padding:30px 0 0 20px;top:0;right:0;bottom:0;left:0;z-index:-1;background:transparent}.a11y-quicknav:focus-within{z-index:100;background:#0003}.a11y-quicknav>button{top:30px;z-index:100;transition:top .22s ease;opacity:0}.a11y-quicknav>button,.a11y-quicknav>button:focus{position:absolute;padding:18px 20px;border-radius:50px;background-color:var(--background-color);line-height:1;border:solid 3px rgb(19 110 248 / 80%);outline:solid 0 #639af9;box-shadow:0 0 0 5px #136ef84d;text-decoration:none;color:var(--text-color);font-size:var(--body-font-size-s);transform:none;text-transform:unset}.a11y-quicknav>button:focus{opacity:1;top:10px;z-index:101}.a11y-quicknav button:after{content:"↵ Enter";font-weight:100;text-transform:uppercase;text-align:center;line-height:.9;border-radius:50px;background-color:#146ff8;padding:5px 10px;color:var(--background-color);margin:0 0 0 1rem;font-size:var(--body-font-size-xxs)}.footer-legal .popup hlx-aria-dialog>button{position:relative;height:auto;width:auto;padding:0;border:none;border-radius:0;color:inherit;font:inherit;z-index:initial;left:auto}footer .popup hlx-aria-dialog>button{height:1px;width:1px;padding:0;margin:0;line-height:0;position:fixed;background-color:transparent!important;top:0;left:-10px;z-index:-1000}@media (prefers-contrast: more){:root{--high-contrast-color: #ab2400;--highlight-color: #ab2400;--color-blue: var(--color-purple);--color-blue-light: var(--color-purple);--color-blue-dark: var(--color-purple);--color-purple-light: var(--color-purple);--color-dark-maroon: var(--color-purple);--color-red: var(--color-purple);--color-gold: var(--color-purple);--color-green: var(--color-purple);--color-magenta: var(--color-purple);--color-mauve: var(--color-purple);--color-orange: var(--color-purple);--color-purple-light-alt: var(--color-purple);--color-orange-transparent: var(--color-purple-transparent);--color-blue-transparent: var(--color-purple-transparent);--color-blue-dark-transparent: var(--color-purple-transparent);--color-purple-light-transparent: var(--color-purple-transparent);--color-red-transparent: var(--color-purple-transparent);--color-gold-transparent: var(--color-purple-transparent);--color-green-transparent: var(--color-purple-transparent);--color-magenta-transparent: var(--color-purple-transparent);--color-mauve-transparent: var(--color-purple-transparent)}.default-content-wrapper a{text-decoration:underline}.nav-sidebar a,.nav-sidebar-links button[aria-controls]:before{--highlight-color: var(--text-color-inverted)}.popular-tags a{--link-color: var(--highlight-color)}}.slick-arrow{position:absolute;padding:0;cursor:pointer;border:none;color:#fff;outline:none;background:var(--color-gray-light-alt);height:48px;width:48px;border-radius:24px;display:block;z-index:99}.slick-prev{left:calc(50% - 80px)}.slick-prev:before,.slick-next:before{font-size:17px;display:inline-block;width:20px;height:1em;background:var(--background-color);content:"";-webkit-mask:url("data:image/svg+xml,%3csvg%20version='1.1'%20id='Layer_1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20x='0px'%20y='0px'%20viewBox='0%200%2019.4%2013'%20style='enable-background:new%200%200%2019.4%2013;'%20xml:space='preserve'%3e%3cpolygon%20fill='currentColor'%20points='0,7.5%2015.5,7.5%2011.7,11.2%2013.1,12.7%2019.4,6.5%2013.1,0.4%2011.7,1.7%2015.5,5.4%200,5.5%20'/%3e%3c/svg%3e") center / 100% 100%;mask:url("data:image/svg+xml,%3csvg%20version='1.1'%20id='Layer_1'%20xmlns='http://www.w3.org/2000/svg'%20xmlns:xlink='http://www.w3.org/1999/xlink'%20x='0px'%20y='0px'%20viewBox='0%200%2019.4%2013'%20style='enable-background:new%200%200%2019.4%2013;'%20xml:space='preserve'%3e%3cpolygon%20fill='currentColor'%20points='0,7.5%2015.5,7.5%2011.7,11.2%2013.1,12.7%2019.4,6.5%2013.1,0.4%2011.7,1.7%2015.5,5.4%200,5.5%20'/%3e%3c/svg%3e") center / 100% 100%;vertical-align:middle}.slick-prev:before{transform:rotate(180deg)}.slick-next{right:calc(50% - 80px)}main .section.search-container.ad-container .ad-wrapper .ad{height:100%}@media (min-width: 600px){main .section[data-background]{background-position:bottom;background-size:100% auto}}@media (min-width: 900px){.section>div{margin:0 auto;width:100%}}@media (min-width: 1200px){.section>div{max-width:1344px}li{margin-bottom:.25rem}}.skeleton{position:relative;height:100%;background-repeat:no-repeat;overflow:hidden}.skeleton.is-animated:after{--width: 300px;position:absolute;content:"";top:0;right:0;bottom:0;left:0;background-repeat:no-repeat;background-image:linear-gradient(-90deg,transparent 0,transparent 25%,white 50%,transparent 75%,transparent 100%);background-size:var(--width) 100%;background-position:0 0;animation:loading 1.5s infinite;rotate:15deg;scale:1 2;translate:calc(-1 * var(--width));will-change:translate}@media (prefers-reduced-motion: no-preference){@keyframes loading{to{translate:100%}}}html>ins[data-google-query-id]{z-index:999!important}.publi-container{padding:1rem}.publi-container:has(iframe){padding:2.5rem}.publi-container-bottom{padding-top:1rem;padding-bottom:1rem}.sub-container,.publi-wrapper{text-align:center;margin:auto}.sub-container{display:flex;justify-content:center;align-items:center}.publi-wrapper-bg:not(:empty){background-color:transparent;border-radius:7px;overflow:hidden}.publi-wrapper-multiplex{width:100%}.social-share-container .publi-container{position:sticky;top:0}.social-share-container .publi-wrapper{margin:0}.modal.hidden{display:none}.error-page{padding:80px 16px;margin:0 auto;display:flex;flex-direction:column;justify-content:center;align-items:center;text-align:center}.error-page h1{font-family:Libre Franklin,Libre Franklin Fallback,sans-serif;color:#000;font-size:28px;line-height:35px}.error-page .error-description{margin-bottom:32px;font-size:20px;line-height:35px}.error-page .action-btns{display:flex;gap:16px}.error-page .action-btns a,.error-page .action-btns button{font-family:Libre Franklin,Libre Franklin Fallback,sans-serif;margin:0;padding:13px 24px;font-size:18px;line-height:22px;font-weight:600}.error-page .reload-btn{background-color:#c74d2f}.error-page .home-btn{display:inline-flex;align-items:center;background-color:transparent;border:1px solid #2F2F33;color:#2f2f33;text-overflow:unset;overflow:visible;justify-content:center;border-radius:30px}@media (min-width: 768px){.error-page{padding-top:120px;padding-bottom:120px;max-width:60%}}*,:before,:after{box-sizing:border-box;border-width:0;border-style:solid;border-color:currentColor}:before,:after{--tw-content: ""}html,:host{line-height:1.5;-webkit-text-size-adjust:100%;-moz-tab-size:4;-o-tab-size:4;tab-size:4;font-family:ui-sans-serif,system-ui,sans-serif,"Apple Color Emoji","Segoe UI Emoji",Segoe UI Symbol,"Noto Color Emoji";font-feature-settings:normal;font-variation-settings:normal;-webkit-tap-highlight-color:transparent}body{margin:0;line-height:inherit}hr{height:0;color:inherit;border-top-width:1px}abbr:where([title]){-webkit-text-decoration:underline dotted;text-decoration:underline dotted}h1,h2,h3,h4,h5,h6{font-size:inherit;font-weight:inherit}a{color:inherit;text-decoration:inherit}b,strong{font-weight:bolder}code,kbd,samp,pre{font-family:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,Liberation Mono,Courier New,monospace;font-feature-settings:normal;font-variation-settings:normal;font-size:1em}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}table{text-indent:0;border-color:inherit;border-collapse:collapse}button,input,optgroup,select,textarea{font-family:inherit;font-feature-settings:inherit;font-variation-settings:inherit;font-size:100%;font-weight:inherit;line-height:inherit;letter-spacing:inherit;color:inherit;margin:0;padding:0}button,select{text-transform:none}button,input:where([type=button]),input:where([type=reset]),input:where([type=submit]){-webkit-appearance:button;background-color:transparent;background-image:none}:-moz-focusring{outline:auto}:-moz-ui-invalid{box-shadow:none}progress{vertical-align:baseline}::-webkit-inner-spin-button,::-webkit-outer-spin-button{height:auto}[type=search]{-webkit-appearance:textfield;outline-offset:-2px}::-webkit-search-decoration{-webkit-appearance:none}::-webkit-file-upload-button{-webkit-appearance:button;font:inherit}summary{display:list-item}blockquote,dl,dd,h1,h2,h3,h4,h5,h6,hr,figure,p,pre{margin:0}fieldset{margin:0;padding:0}legend{padding:0}ol,ul,menu{list-style:none;margin:0;padding:0}dialog{padding:0}textarea{resize:vertical}input::-moz-placeholder,textarea::-moz-placeholder{opacity:1;color:#9ca3af}input::placeholder,textarea::placeholder{opacity:1;color:#9ca3af}button,[role=button]{cursor:pointer}:disabled{cursor:default}img,svg,video,canvas,audio,iframe,embed,object{display:block;vertical-align:middle}img,video{max-width:100%;height:auto}[hidden]{display:none}*,:before,:after{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position: ;--tw-gradient-via-position: ;--tw-gradient-to-position: ;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;--tw-contain-size: ;--tw-contain-layout: ;--tw-contain-paint: ;--tw-contain-style: }::backdrop{--tw-border-spacing-x: 0;--tw-border-spacing-y: 0;--tw-translate-x: 0;--tw-translate-y: 0;--tw-rotate: 0;--tw-skew-x: 0;--tw-skew-y: 0;--tw-scale-x: 1;--tw-scale-y: 1;--tw-pan-x: ;--tw-pan-y: ;--tw-pinch-zoom: ;--tw-scroll-snap-strictness: proximity;--tw-gradient-from-position: ;--tw-gradient-via-position: ;--tw-gradient-to-position: ;--tw-ordinal: ;--tw-slashed-zero: ;--tw-numeric-figure: ;--tw-numeric-spacing: ;--tw-numeric-fraction: ;--tw-ring-inset: ;--tw-ring-offset-width: 0px;--tw-ring-offset-color: #fff;--tw-ring-color: rgb(59 130 246 / .5);--tw-ring-offset-shadow: 0 0 #0000;--tw-ring-shadow: 0 0 #0000;--tw-shadow: 0 0 #0000;--tw-shadow-colored: 0 0 #0000;--tw-blur: ;--tw-brightness: ;--tw-contrast: ;--tw-grayscale: ;--tw-hue-rotate: ;--tw-invert: ;--tw-saturate: ;--tw-sepia: ;--tw-drop-shadow: ;--tw-backdrop-blur: ;--tw-backdrop-brightness: ;--tw-backdrop-contrast: ;--tw-backdrop-grayscale: ;--tw-backdrop-hue-rotate: ;--tw-backdrop-invert: ;--tw-backdrop-opacity: ;--tw-backdrop-saturate: ;--tw-backdrop-sepia: ;--tw-contain-size: ;--tw-contain-layout: ;--tw-contain-paint: ;--tw-contain-style: }.text-primary-color{color:var(--cta-button-color)}:root{font-family:Inter,system-ui,Avenir,Helvetica,Arial,sans-serif;line-height:1.5;font-weight:400;color-scheme:light dark;color:#ffffffde;background-color:#242424;font-synthesis:none;text-rendering:optimizeLegibility;-webkit-font-smoothing:antialiased;-moz-osx-font-smoothing:grayscale}a{font-weight:500;color:#646cff;text-decoration:inherit}a:hover{color:#535bf2}body{margin:0;display:flex;place-items:center;min-width:320px;min-height:100vh}h1{font-size:3.2em;line-height:1.1}button{border-radius:8px;border:1px solid transparent;padding:.6em 1.2em;font-size:1em;font-weight:500;font-family:inherit;background-color:#1a1a1a;cursor:pointer;transition:border-color .25s}button:hover{border-color:#646cff}button:focus,button:focus-visible{outline:4px auto -webkit-focus-ring-color}@media (prefers-color-scheme: light){:root{color:#213547;background-color:#fff}a:hover{color:#747bff}button{background-color:#f9f9f9}}
