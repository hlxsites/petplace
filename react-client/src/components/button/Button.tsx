import { classNames } from "../../util/styleUtil";
import { Icon, IconProps } from "../icon/Icon";
import useButtonBase, { type IUseButtonBase } from "./useButtonBase";

export type IButtonProps = IUseButtonBase &
  JSX.IntrinsicElements["button"] & {
    onClick?: () => void;
  };

const Button = ({
  children,
  className,
  isFullWidth,
  isLoading,
  type = "button",
  variant,
  iconLeft,
  iconRight,
  ...rest
}: IButtonProps) => {
  const { className: baseClassName } = useButtonBase({
    isFullWidth,
    isLoading,
    variant,
  });

  return (
    <button
      className={classNames(baseClassName, className)}
      type={type}
      {...rest}
    >
      {iconLeft && renderIcon(iconLeft, "mr-base")}
      {children}
      {iconRight && renderIcon(iconRight, "ml-base")}
    </button>
  );

  function renderIcon(props: IconProps, className?: string) {
    return <Icon className={className} {...props} />;
  }
};

export default Button;
