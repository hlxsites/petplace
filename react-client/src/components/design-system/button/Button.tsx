import { forwardRef } from "react";
import { classNames } from "~/util/styleUtil";
import { Icon, IconProps } from "../icon/Icon";
import useButtonBase, { type UseButtonBase } from "./useButtonBase";

type ButtonIcon = IconProps["display"];

export type ButtonProps = UseButtonBase &
  JSX.IntrinsicElements["button"] & {
    iconLeft?: ButtonIcon;
    iconProps?: Omit<IconProps, "display">;
    iconRight?: ButtonIcon;
    onClick?: () => void;
  };

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
  (
    {
      children,
      className,
      fullWidth,
      isLoading,
      type = "button",
      variant,
      iconLeft,
      iconProps,
      iconRight,
      ...rest
    },
    ref
  ) => {
    const { className: baseClassName } = useButtonBase({
      fullWidth,
      isLoading,
      variant,
    });

    return (
      <button
        className={classNames(baseClassName, className)}
        ref={ref}
        type={type}
        {...rest}
      >
        {iconLeft && renderIcon({ display: iconLeft, className: "mr-small" })}
        {children}
        {iconRight && renderIcon({ display: iconRight, className: "ml-small" })}
      </button>
    );

    function renderIcon(props: IconProps) {
      return (
        <Icon
          size={16}
          {...props}
          {...iconProps}
          className={classNames(props.className, iconProps?.className)}
        />
      );
    }
  }
);
