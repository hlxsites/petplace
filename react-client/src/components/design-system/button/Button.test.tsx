import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { ComponentProps } from "react";

import { Button } from "./Button";

const { getByRole } = screen;

describe("<Button />", () => {
  it.each(["a-class", "another-class"])(
    `should render %p as a className`,
    (expected) => {
      getRenderer({ className: expected });
      expect(getByRole("button")).toHaveClass(`${expected}`);
    }
  );

  it.each(["Label", "Another label"])(
    "should render button with text '%s'",
    (expected) => {
      getRenderer({ children: expected });
      expect(getByRole("button", { name: expected })).toBeInTheDocument();
    }
  );

  it("should call onClick callback", async () => {
    const onClick = jest.fn();

    getRenderer({ onClick });
    expect(onClick).not.toHaveBeenCalled();

    await userEvent.click(getByRole("button"));
    await userEvent.click(getByRole("button"));
    expect(onClick).toHaveBeenCalledTimes(2);
  });

  it("should have type='button' by default", () => {
    getRenderer();
    expect(getByRole("button")).toHaveAttribute("type", "button");
  });

  it("should have type='submit", () => {
    getRenderer({ type: "submit" });
    expect(getByRole("button")).toHaveAttribute("type", "submit");
  });

  it("should be enabled", () => {
    getRenderer({ disabled: false });

    const button = getByRole("button");
    expect(button).toBeEnabled();
  });

  it("should be disabled", async () => {
    const onClick = jest.fn();

    getRenderer({ disabled: true, onClick });

    const button = getByRole("button");
    expect(button).toBeDisabled();

    await userEvent.click(button);
    expect(onClick).not.toHaveBeenCalled();
  });

  it.each(["the-button-id", "another-button-id"])(
    "button should have id '%s'",
    (expected) => {
      getRenderer({ id: expected });
      expect(getByRole("button")).toHaveAttribute("id", expected);
    }
  );

  it.each(["Custom button", "Another button"])(
    "should have aria-label '%s'",
    (expected) => {
      getRenderer({ "aria-label": expected });
      expect(getByRole("button")).toHaveAttribute("aria-label", expected);
    }
  );

  it("should have base classes", () => {
    getRenderer();

    expect(getByRole("button")).toHaveClass("px-base py-small text-base");
  });

  it("should have primary variant classes", () => {
    getRenderer({ variant: "primary" });

    expect(getByRole("button")).toHaveClass(
      "bg-orange-300-contrast text-white"
    );
  });

  it("should have secondary variant classes", () => {
    getRenderer({ variant: "secondary" });

    expect(getByRole("button")).toHaveClass(
      "bg-white border-neutral-700 text-neutral-700"
    );
  });

  it("should not be full width by default", () => {
    getRenderer();
    expect(getByRole("button")).not.toHaveClass("w-full");
  });

  it("should have full width class", () => {
    getRenderer({ fullWidth: true });
    expect(getByRole("button")).toHaveClass("w-full");
  });
});

// Helpers
type Props = ComponentProps<typeof Button>;
function getRenderer({
  children = <p>Alert text</p>,
  ...rest
}: Partial<Props> = {}) {
  return render(<Button {...rest}>{children}</Button>);
}
