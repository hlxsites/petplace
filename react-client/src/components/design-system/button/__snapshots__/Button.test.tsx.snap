// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot to assure that the component is being rendered correctly 1`] = `
<div>
  <button
    class="rounded-full border-1 font-semibold outline outline-transparent outline-2 outline-offset-2 items-center flex justify-center px-base py-small lg:px-large text-base lg:py-[12px] bg-orange-300-contrast text-white hover:bg-orange-500 focus:bg-orange-300-contrast focus:outline-orange-100"
    type="button"
  >
    <p>
      Alert text
    </p>
  </button>
</div>
`;
